challenge-sizebay/src/Components/ToDoList/ToDoList.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { useContext } from "react";
+import { GlobalContext } from "../../Contexts/GlobalContext";
+import {
+  List,
+  ListItem,
+  RemoveButton,
+  CompleteButton,
+  ButtonsContainer,
+  EmptyListMessage,
+  ClearFilters,
+} from "./style";
+import check from "../../assets/check.png";
+import remove from "../../assets/remove.png";
+import { Tooltip } from "react-tooltip";
+
+const ToDoList = () => {
+  const { states, setters } = useContext(GlobalContext);
+
+  const returnList = () => {
+    return states.listToShow === "pending" ? states.pending : states.done;
+  };
```

Comment:
ref: this can be improved by renaming the method to `getReturnList` and wrapped around a useMemo to avoid extra memory performs.

Reply:
Done! 

useMemo is a hook I've been trying to learn use cases for some time now. I've watched some videos and read some docs, but was still confused. 

The most crucial thing for me to understand it is that I still don't know how to identify multiple renders so that useMemo becomes necessary 🤔  
If you have any tips, I'd reeeally appreciate them :)